snap/squashfs/squashfs.go

```diff
@@ -303,3 +305,33 @@ func (s *Snap) Build(buildDir string) error {
 		).Run()
 	})
 }
+
+// BuildDate returns the "Creation or last append time" as reported by unsquashfs.
+func (s *Snap) BuildDate() time.Time {
+	return BuildDate(s.path)
+}
+
+// BuildDate returns the "Creation or last append time" as reported by unsquashfs.
+func BuildDate(path string) time.Time {
+	var t0 time.Time
+
+	buf, err := exec.Command("unsquashfs", "-s", path).Output()
+	if !(err == nil && bytes.HasPrefix(buf, []byte("Found a valid SQUASHFS"))) {
+		return t0
+	}
+	i := bytes.IndexByte(buf, '\n') + 1
```

Comment:
Personally I would be more comfortable with a simple line based record system where you go line by line and if you see the "Creation or last append time" prefix you do your processing.